### ISSUE TITLE
Duplicate keys in mappings don't cause parse error

### ISSUE THREAD

**Issue by rhysd** [goccy/go-yaml#226]
[YAML spec](https://yaml.org/spec/1.2/spec.html) says at 3.2.1.1:

> The content of a mapping node is an unordered set of key: value node pairs, with the restriction that each of the keys is unique. 

So keys in mappings MUST be unique. This is different from JSON spec where keys SHOULD be unique.

To conform the YAML spec, any YAML parser should cause parse error on duplicate keys. However this library does not cause a parse error.

```go
package main

import (
	"github.com/goccy/go-yaml/parser"
)

func main() {
	s := "foo:\nfoo:\n"
	_, err := parser.ParseBytes([]byte(s), 0)
	if err == nil {
		panic("should have caused parse error")
	}
}
```

Here is a link to playground: https://play.golang.org/p/xog5p3vkQFj

**Comment by marco-m** [comment]
Hello @goccy , I stumbled on the same problem. For my better understanding of the suitability of go-yaml, could you please give us your idea? That is, do you consider this a bug or expected behavior? Thanks!

**Comment by goccy** [comment]
@marco-m @rhysd Thank you for your reports. I plan to fix this.

**Comment by goccy** [comment]
I've fixed this behavior with #531 . I decided to make an error as the default behavior.
